src/shims/posix/thread.rs

```diff
@@ -115,7 +115,24 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 );
             }
 
-            let address = this.read_pointer(&args[1])?;
+            // TODO: Fix this monstrosity
```

Comment:
Why is it even needed?

FWIW, `prctl` in the stdlib has been fixed to be compatible with strict provenance. You shouldn't have to do anything here (and below) any more.

Reply:
That' [this commit](https://github.com/rust-lang/miri/pull/2059/commits/0f1bfa89c822a8bffb3d7fd95e837d897e11bc58), which I accidentally left in tree. Basically, right now the `prctl` code does an implicit transmute of the input arguments to a pointer, which works because of `ptr_from_addr`.

The linked commit tries to make transmuted pointers have invalid provenance, which leads to a enough around `fn` pointers that I'll probably leave it as-is for the time being.

Reply:
> Basically, right now the prctl code does an implicit transmute of the input arguments to a pointer,

That sued to be the case when you started, but has been fixed in the stdlib since then. :)